app/assets/src/components/Samples.jsx

```diff
@@ -2035,10 +2059,23 @@ function TableColumnHeaders({ sort, colMap, filterStatus, state, parent }) {
       <div className="samples-card white">
         <div className="flex-container">
           <ul className="flex-items">
-            <li className="table-header-name">
+            <li className={cs.nameHeader}>
+              {parent.state.displaySelectSamples &&
+                parent.state.readySampleIdsForFilter.length > 0 && (
+                  <div className={cs.checkAllContainer}>
+                    <input
```

Comment:
Is this new?
I wonder if it would be a good time to actually use the Checkbox component.

Reply:
could be a different pr

Reply:
That would be great! I'll do it in a follow-up PR. I did not enjoy using the semantic-ui checkbox.